### PR TITLE
Test ethernomin

### PR DESCRIPTION
**Pull request by zyzek**

### CODE REVIEW

File: contracts/EtherNomin.sol [hunk 1]
```diff
@@ -368,6 +354,28 @@ contract EtherNomin is ERC20FeeToken {
...
+    function canSelfDestruct()
+        public
+        view
+        returns (bool)
+    {
+        // Not being in liquidation implies the timestamp is uint max, so it would roll over.
+        // We need to check whether we're in liquidation first.
+        if (isLiquidating()) {
+            bool totalPeriodElapsed = liquidationTimestamp + liquidationPeriod < now;
+            bool allTokensReturned = (liquidationTimestamp + 1 weeks < now) && (nominPool == totalSupply);
```
Comment (DominicRomanowski) [inline contracts/EtherNomin.sol]: is the 1 weeks necessary?

File: contracts/EtherNomin.sol [hunk 2]
```diff
@@ -456,19 +481,23 @@ contract EtherNomin is ERC20FeeToken {
...
+        uint proceeds;
         if (isLiquidating()) {
-            require(fiatBalanceAllowStale() >= proceeds);
+            proceeds = saleProceedsEtherAllowStale(n);
         } else {
-            require(fiatBalance() >= proceeds);
+            proceeds = saleProceedsEther(n);
         }
+
+        require(this.balance >= proceeds);
```
Comment (DominicRomanowski) [inline contracts/EtherNomin.sol]: couldn't this be: if (this.balance >= proceeds) { proceeds = this.balance }

File: contracts/EtherNomin.sol (hunk 1 above)
Comment (zyzek) [reply contracts/EtherNomin.sol]: Not strictly necessary, but it prevents instant liquidation -> selfdestruction by someone.

File: contracts/EtherNomin.sol (hunk 2 above)
Comment (zyzek) [reply contracts/EtherNomin.sol]: No, we don't allow anyone to be shortchanged. The appropriate response in this case is for the foundation to recollateralise the contract.

**Pull request merged**